TaylorFramework/Modules/scissors/ExtendedComponentExtensions.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  ExtendedComponentExtensions.swift
+//  Taylor
+//
+//  Created by Alexandru Culeva on 4/12/16.
+//  Copyright © 2016 YOPESO. All rights reserved.
+//
+
+import Foundation
+
+extension ExtendedComponent {
+    
+    func perform(@noescape block: ExtendedComponent -> ()) {
+        components.forEach {
+            block($0)
+            $0.perform(block)
+        }
+    }
+    
+    func performReverse(@noescape block: ExtendedComponent -> ()) {
```

Comment:
If you change previous function's name, you'd probably want to change this one's too.